Keep right sidebar open, add Details action

PVince81 opened the pull request:
- [x] TODO: unit tests

@jancborchardt please try it out

Commit:
Keep right sidebar open, add Details action

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/7e497eec-6185-40aa-98cd-219a5c1c04de) was created.

jancborchardt commented:
Please also review @owncloud/designers

PVince81 commented:
Any update on this ? Would be good to getting the interaction sorted out.

PVince81 commented:
Rebased

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d2b0b39c-1612-459a-8d96-005d301ba7f6) was created.

MorrisJobke commented:
```
13:52:00 PhantomJS 1.9.8 (Linux 0.0.0) OCA.Files.FileList tests File actions redisplays actions when new actions have been registered FAILED
13:52:00 	Expected false to equal true.
13:52:00 	    at /var/jenkins/workspace/pull-request-analyser-ng-simple/label/SLAVE/apps/files/tests/js/filelistSpec.js:1993
```

PVince81 commented:
@MorrisJobke I haven't started with the unit tests. I will do so once I get confirmation that this is the way we want to go with the UI interaction.

PVince81 commented:
Any update ? Is this what we want ?
I'll finish this once a decision is made.

MorrisJobke commented:
I would go that way.

Just talked with @tomneedham and it would be awesome to open the sidebar with the three dots click. Then we can but the actions in there. What about this? cc @owncloud/designers

jancborchardt commented:
Why that? Then any other action requires a lot of mental work to find. Also the interface would switch around when you don’t really expect it.

(Btw we really need a short animation for the appearance of the sidebar.)

MorrisJobke commented:
> Why that? Then any other action requires a lot of mental work to find. Also the interface would switch around when you don’t really expect it.

Because then it's also a two step thing to do everything. And nearly all actions (beside rename and download) will open the sidebar anyways.

MorrisJobke commented:
This was just a thought.

jancborchardt commented:
> nearly all actions (beside rename and download) will open the sidebar anyways.

See it the other way around: Actually _only_ the Versions action from the actions dropdown require the sidebar. The other 3 (Rename, Download and Delete) do not. ;)

MorrisJobke commented:
> See it the other way around: Actually only the Versions action from the actions dropdown require the sidebar. The other 3 (Rename, Download and Delete) do not. ;)

I think both sides are valid and I'm fine with this. -> :+1: but a rebase is needed.

@jancborchardt do we want to do this? I think so.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/da85a9b7-366f-45d3-b4f4-1cfcc7728ea9) was created.

PVince81 commented:
Rebased.

Should I go ahead and finish this then ?

MorrisJobke commented:
> Should I go ahead and finish this then ?

@jancborchardt Please answer here

Commit:
Fixed sidebar toggle entry

Properly highlight the row after selecting the item.
This also fixes the unit tests by changing the order of registration of
file actions.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/e4563250-afa5-4f15-b31e-ffc1448783a3) was created.

PVince81 commented:
Fixed this PR for the unit tests.

This is ready for review @jancborchardt @MorrisJobke @owncloud/designers @nickvergessen @schiesbn

MorrisJobke commented:
Tested and works :+1:

@jancborchardt Can you provide an icon for this?

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/38e8750e-16ac-4ce2-9fa1-ac6ba6b006ca) was created.

jancborchardt commented:
I added an icon., and it works good. One remark though: Please put the Details action as the very first item in the actions list. The order should be like this:

* Details
* Download
* Rename
* Versions
* Delete

PVince81 commented:
Dammit... once again an order issue. This depends on the loading order of the apps.
I know you hate it that plugins can specify a sort order but I don't see how we can solve this without hard-coding id strings in core, which is even worse.

PVince81 commented:
General ordering problem discussed here: https://github.com/owncloud/core/issues/18280#issuecomment-137346384

jancborchardt commented:
I think hard-coding the order in core is way better than again allowing apps to set their own arbitrary order parameter, which has been shown to fail with the app order.

karlitschek commented:
Yes. We should add a detail item as discussed.

oparoz commented:
Just add 10 to the order set by an app if you can make the difference between 3rd party and core apps and set the rest in stone.

jancborchardt commented:
Well, delete should always be at the end. The other actions can be sorted alphabetically for example. I don’t see how introducing an app-defined order which devs need to set solves anything we can’t do here.

oparoz commented:
@jancborchardt - Actually categories instead of specific order might be useful. Group file actions together, same for viewers/editors, etc.

karlitschek commented:
ping. can be merge this?

schiesbn commented:
Looks good to me. Only one small detail (but this could also be addressed in a separate bug fix). If the sidebar is open and I click on a empty space below the files list the preview icon and file name is gone but the "versions" and "sharing" tabs are still there which looks quite strange. Either we should close the side bar in this case or keep the last selected file selected until you chose a different file.

PVince81 commented:
Rebased.

I'll look into fixing the order stuff separately (tabs order and menu item order).

karlitschek commented:
:+1:

schiesbn commented:
:+1: beside the small things I mentioned here https://github.com/owncloud/core/pull/18273#issuecomment-140036095, maybe you can address them during the bug fix phase.

nickvergessen commented on apps/files/js/filelist.js:
```diff
@@ -285,11 +290,25 @@
+
+			if (this._detailsView) {
+				this.fileActions.registerAction({
+					name: 'Details',
```
not translated?

PVince81 replied:
It should internally call t() with that. And now I realize that it cannot work.

I'll make a PR to add it as displayName (`name` is actually the action id, legacy stuff...)

PVince81 replied:
@nickvergessen https://github.com/owncloud/core/pull/19266
